Review for task "weatherman" - Mazahar

mazahar02 opened the pull request:
Description:
Please review my code
Thanks

mazahar02 committed:
i tried

mazahar02 committed:
Merge pull request #9 from wadoodislam/mazahar-master

tried

mazahar02 committed:
Add files via upload

I have done all up to my extent of knowledge 
sir please have a look and instruct further 
thanks

mazahar02 committed:
Add files via upload

mazahar02 committed:
Add files via upload

mazahar02 committed:
Add files via upload

mazahar02 committed:
moved files in folder

mazahar02 committed:
Sir i corrected the haphazardness 

Sir i created the class for declaration of variables and its producing the accurate result
It is generating the reports for a complete year
Please instruct me further so i can join all the modules in a file

mazahar02 committed:
Complete first module

Sir this file is checked and working fine with command line as well
it is producing accurate result according to your given instruction like this one
weatherman.py /path/to/files-dir -e 2002
its call should be like
cmd_chk.py [address of the weather files]+[ year you want to report of ]
please have a look and instruct further

mazahar02 committed:
Sir this is a changed module have a look

Sir i gave the code a presentable shape
the code is producing accurate results
your second command 
weatherman.py /path/to/files-dir -a 2005/6
its working fine over command prompt with same command pattern
recreated_months.py [address of files] + [year number / month number]
please have a look and instruct further

mazahar02 committed:
Add files via upload

mazahar02 committed:
Sir the first 3 tasks

Sir i concatenated the first 3 tasks into one file
this file is working perfect over command prompt
the command to run this file is same as you mentioned in the task
first
python { location of files ] + [ arguments to be passed }
only in creating a horizontal bar graph you would need to add a word 'bar' in the command like this
python [ location of files ] + [ 'bar' } + { argument to be passed }
i added this 'bar' word in order to differentiate between the monthly report and monthly bar_graph report because as per the task there is no difference so the interpreter will fail
This is one file for first three tasks and you may differ in arguments with your choice 
The code itself figures it out what is the desired result
I'll look forward for your time
Thanks

mazahar02 committed:
Sir please consider

Sir i have assembled all the modules in one file 
The file is ready to be examined
You just need to run it over cmd 
path of the backup files is all it needs
then any argument further given to it 
It configures himself what would be the desired result
You may please have a look and instruct further 
Thanks

wadoodislam commented on weatherman/All_tasks.py:
```diff
@@ -0,0 +1,292 @@
+import os
+import sys
+
+class readings():
```
Class names are always title cased.

wadoodislam commented on weatherman/All_tasks.py:
```diff
@@ -0,0 +1,292 @@
+import sys
+
+class readings():
+    highest_temp = 0
```
let's not make static variables and set these field in `__init__` so that new objects of this class doesn't discard previous values.

wadoodislam commented on weatherman/All_tasks.py:
```diff
@@ -0,0 +1,292 @@
+
+
+
+obj = readings
```
let's not define unnecessary variables

wadoodislam commented on weatherman/All_tasks.py:
```diff
@@ -0,0 +1,292 @@
+def main():
+
+    while True:
+        if len(sys.argv) > 3 and sys.argv[2] != 'bar' :
```
use `argparse` package to handle the commandline arguments.

wadoodislam commented on weatherman/All_tasks.py:
```diff
@@ -0,0 +1,292 @@
+
+def main():
+
+    while True:
```
let's removed this loop

wadoodislam commented on weatherman/yearly_gui.py:
```diff
@@ -0,0 +1,146 @@
+import tkinter as tk
+
+
+root=Tk()
```
Removed GUI from your project. it is not required in the task.

wadoodislam commented on weatherman/All_tasks.py:
```diff
@@ -0,0 +1,292 @@
+    date4 = ''
+
+
+def years(year_no):
```
Let's defined a new class here which populated and handles a list of object of `Reading` based on what arguments were passed through the command line. Also, use glob and csv.Dictreader to read the matched files from the directory containing the weatherfiles.

wadoodislam reviewed (changes requested):
Please Remove GUI from your project. Let's improve the architecture first.

mazahar02 committed:
Sir i fixed all possible shortcomings 

Sir 
i have carved the best possibility according to the directed instruction
+ class name is title cased
+ changed static variables 
+ made dynamic encapsulated in (__init__) function
+ cleared all unnecessary declarations
+ removed the main() loop
+ created separate class for readings and separate function for output
+ both functions are independent 
+ i am not using the GUI in this commit 
+ i have used csv Dictreader as directed
+ There arouse an error while using glob as matching name because file address uses "\" in address and we cannot split data over "\"
Rest everything is working perfect 
code is some how in an acceptable shape
Kindly instruct further as desired 
Thanks
